python/paddle/incubate/autograd/primx.py

```diff
@@ -0,0 +1,611 @@
+                    expand_nested_list(to_tensors(lower_fn(op, *input_args)))):
+                assert not (orig_out is None) ^ (
+                    new_out is None), "orig_out and new_out should match."
+                vars_to_remove.add(new_out.name)
+                value_table[new_out.name] = new_out
+                to_bind[orig_out.name] = new_out.name
+                to_bind_rev[new_out.name] = orig_out.name
+        else:
+            inputs = {}
+            for i in range(len(op.input_names)):
+                inputs[op.input_names[i]] = bind_name(
+                    op.input(op.input_names[i]), to_bind)
+
+            outputs = {}
+            for i in range(len(op.output_names)):
+                outputs[op.output_names[i]] = op.output(op.output_names[i])
+
+            attrs = {}
+            for name in sorted(op.attr_names):
+                attrs[name] = op.attr(name)
+            from paddle.fluid.dygraph.base import param_guard
+            new_op_desc = block.desc.append_op()
+            with param_guard(inputs), param_guard(outputs):
+                op = Operator(
+                    block=block,
+                    desc=new_op_desc,
+                    type=op.type,
+                    inputs=inputs,
+                    outputs=outputs,
+                    attrs=attrs)
+            block.ops.append(op)
+
+    # Step3: Do some post-processing work
+    for op_idx in reversed(ops_to_remove):
+        block.desc._remove_op(op_idx, op_idx + 1)
+        del block.ops[op_idx]
+    block._sync_with_cpp()
+
+    for op_idx in range(len(block.ops)):
+        op = block.ops[op_idx]
+        for in_name in op.input_arg_names:
+            if in_name in to_bind_rev:
+                op._rename_input(in_name, to_bind_rev[in_name])
+
+        for out_name in op.output_arg_names:
+            if out_name in to_bind_rev:
+                op._rename_output(out_name, to_bind_rev[out_name])
+
+    for var_name in sorted(vars_to_remove):
+        assert var_name in to_bind_rev, 'var_name "{}" is not in to_bind_rev.'.format(
+            var_name)
+        if var_name != to_bind_rev[var_name]:
+            block.desc._remove_var(cpt.to_bytes(var_name))
+            del block.vars[var_name]
+    block._sync_with_cpp()
+
+
+@framework.static_only
+def orig2prim(block=None):
+    """ 
+    .. note::
+        **This API is ONLY available in the static mode.**
+
+    All operators in the target block are processed as follows.
+    If it is an original operator, it will be transformed into
+    one or a series of automatic differential basic operators with
+    equivalent function.
+    
+    Args:
+        block(paddle.fluid.framework.Variable|None, optional): The
+            target block to process on. Default None, and will
+            process on the current block of main program.
+    
+    Returns:
+        None
+    """
+    _lower(block, reverse=False)
+
+
+@framework.static_only
+def prim2orig(block=None):
```

Comment:
~~This API is~~ ONLY available in the static mode.

Reply:
updated